### DISCUSSION TITLE
Nested query with where clause

### DISCUSSION THREAD

**Discussion by satya1395** [prisma/prisma#19777] [Q&A]
how can I pull all posts from channel that user follows. 
The relation is like - user follows multiple channels, each channel can have multiple posts.

```prisma
model Channel {
  id            String          @id @default(uuid())
  title         String
  bio           String
  followerCount Int             @default(0) @map(name: "follower_count")
  posts         Post[]
  followers     UserOnChannel[]
}

model Post {
  id          String   @id @default(uuid())
  title       String
  content     String   @db.VarChar(500)
  likes       Int      @default(0)
  imageUrl    String   @map(name: "image_url")
  channelId   String   @map(name: "channel_id")
  publishedAt DateTime @map(name: "published_at")
  processedAt DateTime @default(now()) @map(name: "processed_at")
  channel     Channel  @relation(fields: [channelId], references: [id])
}


model User {
  id        String          @id @default(uuid())
  name      String?
  email     String?         @unique
  createdAt DateTime        @default(now()) @map(name: "created_at")
  updatedAt DateTime        @updatedAt @map(name: "updated_at")
  following UserOnChannel[]
  @@map(name: "users")
}
```

**Comment by ludralph** [comment]
Hi @satya1395 👋 

From the schema you shared, the `UserOnChannel` model is missing but i have assumed the model for `UserOnChannel` looks like this:
```prisma
model UserOnChannel {
  id        String   @id @default(uuid())
  userId    String   @map(name: "user_id")
  channelId String   @map(name: "channel_id")
  createdAt DateTime @default(now()) @map(name: "created_at")
  user      User     @relation(fields: [userId], references: [id])
  channel   Channel  @relation(fields: [channelId], references: [id])
}
```

To pull all posts from channels that a user follows, you can write a function as shown below:
```js
const getPostsFromFollowedChannels = async (userId) => {
  const user = await prisma.user.findUnique({
    where: { id: userId },
    include: { following: { include: { channel: { include: { posts: true } } } } },
  });

  if (!user) {
    throw new Error(`User with ID ${userId} not found.`);
  }

  const posts = user.following.reduce((acc, userOnChannel) => {
    return acc.concat(userOnChannel.channel.posts);
  }, []);

  return posts;
}
```
The `getPostsFromFollowedChannels` function uses Prisma's `findUnique` method to fetch the user by ID. It includes the `following` relation with the `channel` and its associated `posts` using nested includes. It then flattens the posts from all followed channels into a single array and returns it.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ludralph** [reply]
Hey @satya1395,

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by nurul3101** [reply]
@satya1395 Please let us know in case this issue is resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
